ngx_http_sxg_filter_module.c

```diff
@@ -267,18 +287,35 @@ static bool calc_integrity(const ngx_http_request_t* const req,
 
 static ngx_int_t subresource_fetch_handler(ngx_http_request_t* req, void* data,
                                            ngx_int_t rc) {
-  ngx_http_sxg_ctx_t* ctx = data;
+  if (req->done) {
+    return NGX_OK;
+  }
   ngx_http_core_srv_conf_t* cscf =
       ngx_http_get_module_srv_conf(req, ngx_http_core_module);
-  if (req->done) {
+  ngx_http_sxg_ctx_t* ctx = data;
+  if (ctx == NULL || ctx->finished == true) {
     return NGX_OK;
   }
-  if (req->upstream->headers_in.status_n != 200) {
+
+  if (pthread_mutex_lock(&ctx->lock) != 0) {
```

Comment:
Should the check of `ctx->finished` be protected by `ctx->lock`?  Repeated below.

Reply:
Oh, I realized that I can use `req->done` for this purpose. It does not requires mutex.